test_pull_requests

```diff
@@ -1587,23 +1587,54 @@ popd
       URI(build_url + "/api/json")
     end
 
-    def self.is_previous_build_running?(branch, settings)
+    def self.previous_build_running?(branch, settings, repo)
+      addtl_jenkins_params = branch_settings(branch, settings)['addtl_jenkins_params']
       build_num = get_next_build(branch, settings) - 1
       build_url = Properties['jenkins_url'] + 'job/' + branch_settings(branch, settings)['jenkins_job_name'] + "/#{build_num}/"
+      previous_running = false
+      consecutive_not_running_count = 0
       get_session(branch, settings).start do |http|
-        request = Net::HTTP::Get.new to_api_url(build_url).request_uri
-        response = http.request request
-        if response.kind_of? Net::HTTPSuccess
-          return JSON.parse(response.body)["building"]
-        elsif response.kind_of? Net::HTTPNotFound
-          return false # maybe someone deleted the last build?
-        else
-          return true # assume it's running to avoid errors causing multiple builds to start
+        # Search for a running build of the same variety.  Don't run forever.
+        100.times do
+          # Break if you found one or if you run across a bunch of consecutive builds that aren't running you can assume you reached the end
+          break if previous_running || consecutive_not_running_count >= 10
+          request = Net::HTTP::Get.new to_api_url(build_url).request_uri
+          response = http.request request
+          if response.kind_of? Net::HTTPSuccess
+            previous_running = build["building"]
+            build_json = JSON.parse(response.body)
+            if previous_running
+              consecutive_not_running_count = 0
+              submitted_params = build_json['actions']['parameters']
+              # Validate that the params match between the test settings and jenkins for cases that the same job is being used for multiple test commands
+              if submitted_params && addtl_jenkins_params && addtl_jenkins_params[repo]
+                submitted_param_map = {}
+                submitted_params.each do |param|
+                  submitted_param_map[param['name']] = param['value']
+                end
+                addtl_jenkins_params[repo].each do |key, value|
+                  next if value == '*'
+                  submitted_value = submitted_param_map[key]
+                  if submitted_value && submitted_value != value
+                    previous_running = false
```

Comment:
This doesn't make sense -- we found a build that didn't match, but it was running -- why do we want to move on to the next one and assume this build wasn't running?